docker-compose.yml

```diff
@@ -1,38 +1,50 @@
-osm_data:
-  image: debian:jessie
-  command: echo just a data container
-  volumes:
-    - osm-world_osm_data:/var/data/osm-planet
-
-osm_cache:
-  image: debian:jessie
-  command: echo just a data container
-  volumes:
-    - osm-world_osm_cache:/var/cache/osm-cache
-
-postgis:
-  image: geometalab/postgis:9.5
-  volumes:
-    - osm_world-db-postgis-data:/var/lib/postgresql
-  command: postgres -B 12GB -F -N 512 -S 20GB
-  environment:
-    - POSTGRES_DB=gis
-    - POSTGRES_USER=gis
-    - POSTGRES_PASSWORD=gis
-  log_opt:
-    max-size: 50k
-
-osm_importer:
-  image: geometalab/osm2pgsql:latest
-  # for development, disable the line above and uncomment the line below
-  # build: .
-  command: /root/run.sh
-  links:
-    - postgis:database
-  volumes_from:
-    - osm_data
-    - osm_cache
-  environment:
-    - num_processes=16
-  log_opt:
-    max-size: 50k
+version: '2'
+services:
+  database:
+    # to limit scaling to exactly 1, we need to specify the container name
+    container_name: osmworlddb_postgis
+    image: geometalab/postgis:9.5
+    volumes:
+      - database-postgis-data:/var/lib/postgresql
+    command: postgres -B 12GB -F -N 512 -S 20GB --max_wal_size=5GB
+    environment:
+      - POSTGRES_DB=gis
+      - POSTGRES_USER=gis
+      - POSTGRES_PASSWORD=gis
+    networks:
+      - osm-world-intern
+      - osm-world
+    logging:
+      driver: "json-file"
+      options:
+        max-size: 50k
+  osm_importer:
+    # to limit scaling to exactly 1, we need to specify the container name
+    container_name: osmworlddb_osm_importer
+    image: geometalab/osm2pgsql:latest
+    # for development, disable the line above and uncomment the line below
+#    build: .
+    entrypoint: /root/wait-for-it.sh database:5432 --
+    command: /root/run.sh
+    depends_on:
+      - database
+    volumes:
+      - osm_data:/var/data/osm-planet
+      - osm_cache:/var/cache/osm-cache
+    environment:
+      - num_processes=16
+    networks:
+      - osm-world-intern
+    logging:
+      driver: "json-file"
+      options:
+        max-size: 50k
+volumes:
+  osm_data: {}
+  osm_cache: {}
+  database-postgis-data: {}
+networks:
+  osm-world:
+    driver: bridge
+  osm-world-intern:
```

Comment:
This has nothing to do with a trainee, so it should probably be named '&hellip;-internal' instead of '&hellip;-intern'.